libp2p.go

```diff
@@ -180,13 +180,19 @@ func newWithCfg(ctx context.Context, cfg *Config) (host.Host, error) {
 		ps = pstore.NewPeerstore()
 	}
 
+	// Set default muxer if none was passed in
+	muxer := cfg.Muxer
```

Comment:
for consistency with the code below, it would be better to set it in the `cfg` object directly:
```go
if cfg.Muxer == nil {
   cfg.Muxer = DefaultMuxer()
}
```

Reply:
Hm. But that would modify the user's config. I prefer it this way.